src/types/local_date.rs

```diff
@@ -0,0 +1,290 @@
+
+impl Ymd {
+    /// Counts the number of days since 01.01.1970.
+    fn to_days(&self) -> i32 {
+        let mut total_days = 0;
+
+        let number_day = DAY_IN_MONTH[self.month as usize - 1] // number of days from 1 January
+            + if is_leap_year(self.year) && self.month > 2 {
+                0
+            } else {
+                -1
+            }
+            + self.day as i32;
+
+        if self.year >= 1970 {
+            total_days += (self.year - 1970) * 365 + number_leap_years(self.year - 1)
+                - LEAP_YEAR_1970
+                + number_day;
+        } else {
+            total_days -= if is_leap_year(self.year) { 366 } else { 365 } - number_day;
+            if self.year < 0 {
+                total_days -= (1970 - self.year - 1) * 365
+                    + number_leap_years(self.year.abs() + 1)
+                    + LEAP_YEAR_1970
+                    + 1;
+            } else {
+                total_days -= (1970 - self.year - 1) * 365 + LEAP_YEAR_1970
+                    - number_leap_years(self.year + 1);
+            }
+        }
+        total_days
+    }
+
+    /// Create a Ymd from the number of days since 01.01.1970.
+    fn from_days(mut n: i64) -> Option<Self> {
+        if !(MIN_JS_DATE..=MAX_JS_DATE).contains(&(n as i32)) {
+            None
+        } else {
+            let mut year = 1970;
+            while n.abs() >= 365 {
+                // Find the number of years in n days.
+                let k = (n / 365) as i32;
+                if year > 0 && year + k < 0 {
+                    n += 1;
+                }
+                year += k;
+
+                if k > 0 {
+                    n -= (k.abs() * 365 + number_leap_years(year - 1) - number_leap_years(year - k))
+                        as i64;
+                } else {
+                    n += (k.abs() * 365 + number_leap_years(year - k) - number_leap_years(year + 1))
+                        as i64;
+                }
+            }
+
+            if n < 0 {
+                // If the remaining number of days is negative, change the year and count the complement.
+                year -= 1;
+                n += if is_leap_year(year) { 366 } else { 365 };
+            }
+
+            if is_leap_year(year) && n >= 60 {
+                n -= 1;
+            } else if is_leap_year(year) && n > 31 {
+                return Some(Ymd {
+                    year,
+                    month: 2,
+                    day: (n - 30) as i8,
+                });
+            }
+
+            let q = DAY_IN_MONTH // Find month.
+                .iter()
+                .enumerate()
+                .filter(|&(_, &x)| i64::from(x) <= n)
+                .max_by_key(|&(_, &x)| x)
+                .map(|(index, &_value)| index + 1)
+                .unwrap();
+
+            Some(Ymd {
+                year,
+                month: q as i8,
+                day: (n - DAY_IN_MONTH[q - 1] as i64) as i8 + 1,
+            })
+        }
+    }
+}
```

Comment:
As this is a very nontrivial logic, please add a suite of unit tests for both functions, preferably in this file (in a dedicated `mod tests`).